nextjs/packages/next/shared/lib/router/router.ts

```diff
@@ -269,12 +269,17 @@ function omitParmsFromQuery(query: ParsedUrlQuery, params: string[]) {
   return filteredQuery
 }
 
+export function resolveHrefOnServer(href: Url): string {
```

Comment:
Changes in this file can be reverted if we use `formatWithValidation` in `normalizeRedirectValues` function